python/pyspark/sql/functions/builtin.py

```diff
@@ -1271,6 +1271,11 @@ def max_by(col: "ColumnOrName", ord: "ColumnOrName") -> Column:
     .. versionchanged:: 3.4.0
         Supports Spark Connect.
 
+    Notes
+    -----
+    The function is non-deterministic when the maximum value from `ord` is associated
+    with multiple values from `col`.
```

Comment:
I think this is a bit hard to understand. Is there a simpler example we can use to demonstrate this idea?

Reply:
I think we can just say that `` The function is non-deterministic so the output order can be different for those associated the same values of `col ` ``